### PR TITLE
Added Simmer Index and Simmer Zone

### PR DESCRIPTION
**Pull request by ahmadtawakol**
Let's try this again! I messed up the [previous PR](https://github.com/dolezsa/thermal_comfort/pull/65) and thought it would be easier to create a new one.

This is based on another fork of thermal_comfort by @lymanepp at https://github.com/lymanepp/thermal_comfort. I adapted the code to work with this plugin. The only thing missing is the translations for the Simmer Zone strings.

### CODE REVIEW

File: custom_components/thermal_comfort/sensor.py [hunk 1]
```diff
@@ -255,6 +267,38 @@ def computeRiskLevel(self, temperature, humidity):
...
 
+    def computeSimmerIndex(self, temperature, humidity):
+        """https://www.vcalc.com/wiki/rklarsen/Summer+Simmer+Index"""
+        fahrenheit = util.temperature.celsius_to_fahrenheit(temperature)
+
+        si = (1.98 * (fahrenheit - (0.55 - (0.0055 * humidity)) * (fahrenheit - 58.0)) - 56.83)
+
+        if fahrenheit < 70:
+            si = fahrenheit
+
```
Comment (rautesamtr) [inline custom_components/thermal_comfort/sensor.py:278]: Theres some whitespace here when viewing the commit localy.

File: custom_components/thermal_comfort/sensor.py [hunk 2]
```diff
@@ -255,6 +267,38 @@ def computeRiskLevel(self, temperature, humidity):
...
         return 0  # No risk of frost
 
+    def computeSimmerIndex(self, temperature, humidity):
+        """https://www.vcalc.com/wiki/rklarsen/Summer+Simmer+Index"""
+        fahrenheit = util.temperature.celsius_to_fahrenheit(temperature)
+
+        si = (1.98 * (fahrenheit - (0.55 - (0.0055 * humidity)) * (fahrenheit - 58.0)) - 56.83)
+
+        if fahrenheit < 70:
+            si = fahrenheit
```
Comment (rautesamtr) [inline custom_components/thermal_comfort/sensor.py:277]: Tests are broken since you did not consider this.

**Review by rautesamtr — CHANGES_REQUESTED** [review]
Hey, thank you for your effort so far. I appreciate it. 

Since contributor documentation is still non existent, here some tipps.

You can use [pre-commit](https://pre-commit.com/) to run some checks before commiting.

To run the tests locally, run the following in a bash shell in the projects root directory:
```bash
python -m venv venv
source venv/bin/activate
pip install -r requirements_test.txt
pytest
```

For the failing `assert len(hass.states.async_all(sensor.DOMAIN))` you could just fix the numbers, or add a constant eg `LEN_TEST_SENSORS = len(DEFAULT_SENSOR_TYPES) + 2` (Total default sensors + 2 command line input sensors) and use that for the checks with a coresponding offset where needed.

File: custom_components/thermal_comfort/sensor.py (hunk 1 above)
Comment (rautesamtr) [reply custom_components/thermal_comfort/sensor.py:278]: image

**Comment by ahmadtawakol** [comment]
Sorry about that! I fixed the problems you pointed out. Let me know if there is anything else I missed or need to do!

**Review by rautesamtr — APPROVED** [review]
Looks good to me. Thank you!

**Comment by lymanepp** [comment]
No credit will be given to the author?

**Comment by ahmadtawakol** [comment]
@lymanepp You are mentioned in the release notes as well as a co-author on the commit in this PR.

**Comment by rautesamtr** [comment]
image
image

**Comment by lymanepp** [comment]
> @lymanepp You are mentioned in the release notes as well as a co-author on the commit in this PR.

I want a permanent mention in the README, not the PR or release notes that will soon be forgotten.

**Comment by rautesamtr** [comment]
None of us has a permanent mention in the README. I explicitly askef if it was enough to be added as co author of the commit. If this is that important for you we can start a CONTRIBUTORS file and add you.

**Comment by lymanepp** [comment]
> If this is that important for you we can start a CONTRIBUTORS file and add you.

That's a good idea!

**Pull request merged**